File: builder/tmpl/func/func.js
{{define "js"}}

// User-defined function
var main = {{code}}

// Invoke function with JSON (un)loading
try {
  var res = main(JSON.parse(process.argv[2]));
  var buf = Buffer.from(JSON.stringify(res));
  process.stdout.write(buf);
} catch (err) {
  var buf = Buffer.from(err);
  process.stderr.write(buf);
  process.exit(1);
}

{{end}}

File: builder/tmpl/func/func.py
{{define "py"}}
import sys
import rapidjson as json

# user-defined function
main = {{code}}

if __name__ == '__main__':
   # Invoke user-defined func with JSON (un)loading
    try:
        ctx = json.loads(sys.stdin.buffer.read().decode())
        res = main(ctx)
        sys.stdout.buffer.write(json.dump(res))
    except Exception as e:
        sys.stderr.buffer.write(str(e).encode())
        exit(1)

{{end}}